foundations/src/lib.rs

```diff
@@ -69,7 +69,7 @@ mod utils;
 #[cfg(feature = "cli")]
 pub mod cli;
 
-#[cfg(feature = "settings")]
+#[cfg(any(feature = "settings", feature = "telemetry-server"))]
```

Comment:
Let's put `ListenAddr` elsewhere, rather than expose this module when `settings` is off.